internal/controller/rabbitmq/rabbitmq_controller.go

```diff
@@ -623,25 +631,59 @@ func (r *Reconciler) deletePerPodServices(ctx context.Context, instance *rabbitm
 	return nil
 }
 
-func updateMirroredPolicy(ctx context.Context, helper *helper.Helper, instance *rabbitmqv1beta1.RabbitMq, config *rest.Config, apply bool) error {
-	cli := helper.GetKClient()
-
-	pod := types.NamespacedName{
-		Name:      instance.Name + "-server-0",
+func updateMirroredPolicy(ctx context.Context, helper *helper.Helper, instance *rabbitmqv1beta1.RabbitMq, apply bool) error {
+	policyName := types.NamespacedName{
+		Name:      instance.Name + "-ha-all",
 		Namespace: instance.Namespace,
 	}
 
-	container := "rabbitmq"
-	s := []string{"/bin/bash", "-c", "rabbitmqctl clear_policy ha-all"}
+	policy := &rabbitmqv1beta1.RabbitMQPolicy{}
+	err := helper.GetClient().Get(ctx, policyName, policy)
+
+	if apply && k8s_errors.IsNotFound(err) {
+		// Create the policy CR
+		definition := map[string]interface{}{
+			"ha-mode":                "exactly",
+			"ha-params":              2,
+			"ha-promote-on-shutdown": "always",
+		}
+		definitionJSON, marshalErr := json.Marshal(definition)
+		if marshalErr != nil {
+			return marshalErr
+		}
+
+		policy = &rabbitmqv1beta1.RabbitMQPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      policyName.Name,
+				Namespace: policyName.Namespace,
+			},
+			Spec: rabbitmqv1beta1.RabbitMQPolicySpec{
+				RabbitmqClusterName: instance.Name,
+				Name:                "ha-all",
+				Pattern:             "",
+				Definition:          apiextensionsv1.JSON{Raw: definitionJSON},
+				Priority:            0,
+				ApplyTo:             "all",
+			},
+		}
+		if err := controllerutil.SetControllerReference(instance, policy, helper.GetScheme()); err != nil {
+			return err
+		}
+		return helper.GetClient().Create(ctx, policy)
+	}
 
 	if apply {
-		s = []string{"/bin/bash", "-c", "rabbitmqctl set_policy ha-all \"\" '{\"ha-mode\":\"exactly\",\"ha-params\":2,\"ha-promote-on-shutdown\":\"always\"}'"}
+		// Policy already exists, nothing to do
+		return err
 	}
 
-	err := rsh.ExecInPod(ctx, cli, config, pod, container, s,
-		func(_ *bytes.Buffer, _ *bytes.Buffer) error {
-			return nil
-		})
+	// Delete the policy CR if it exists
+	if err == nil {
+		return helper.GetClient().Delete(ctx, policy)
+	}
+	if k8s_errors.IsNotFound(err) {
+		return nil
+	}
```

Comment:
not sure I understand why we delete the policy here. is the reason if we call the func with apply == false we want it to be deleted? maybe it is cleaner to have a createPolicy and deletePolicy func.